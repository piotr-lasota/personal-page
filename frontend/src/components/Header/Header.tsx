import React, { useCallback, useState } from 'react';
import { Button } from 'gatsby-theme-material-ui';

type HeaderProps = {
  title: string;
};

const Header = ({ title }: HeaderProps): JSX.Element => {
  const [showSubtext, setShowSubtext] = useState(false);

  const handleShowSubtextClicked = useCallback(
    () => setShowSubtext(true),
    [setShowSubtext]
  );

  return (
    <header>
      <p>{title}</p>
      <Button onClick={handleShowSubtextClicked}>Show subtext</Button>
      {showSubtext && <p>This is a subtext</p>}
    </header>
  );
};

export default Header;
