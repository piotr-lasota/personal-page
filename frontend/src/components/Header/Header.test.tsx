/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import Header from './Header';

describe('Header', () => {
  it('renders correctly', () => {
    const { baseElement } = render(<Header title="Default Starter" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Passes', () => {
    // Arrange
    const value = 3;

    // Act & Assert
    expect(value).toBe(3);
  });

  it('Shows the subtext after the button click', async () => {
    // Arrange
    const title = 'My title';
    render(<Header title={title} />);

    // Act
    fireEvent.click(screen.getByText('Show subtext'));
    await waitFor(() => screen.getByText('This is a subtext'));

    // Assert
    expect(screen.getByText('This is a subtext')).toBeInTheDocument();
  });
});
