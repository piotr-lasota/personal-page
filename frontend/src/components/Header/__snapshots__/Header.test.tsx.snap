// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<body>
  <div>
    <header>
      <p>
        Default Starter
      </p>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Show subtext
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </header>
  </div>
</body>
`;
